localstack/services/apigateway/helpers.py

```diff
@@ -138,7 +151,20 @@ def _resolve_refpath(self, refpath: str) -> dict:
         if refpath in self._refpaths and not self.allow_recursive:
             raise Exception("recursion detected with allow_recursive=False")
 
-        if refpath in self._cache:
+        # We don't resolve the Model definition, we will return a absolute reference to the model like AWS
```

Comment:
it can be a bug, sure, but the cache should avoid all circular loops. Resolving all refs up front, cleans up a lot of the lookups after.

Reply:
Agreed. However, AWS does not resolve the Model at all as shown by the snapshot, they replace it with an absolute reference to the Model. This is also how you set up reference to other `Model` with the API outside of importing. I'd say `Model` is a specific case here.

Also, not resolving Model makes sense, because we don't use the schema of the `Model` but only set its name to the `requestModels` and `responseModels`.

Reply:
Also, the JSON Schema spec is talking about recursion and circular references, and they say that it's disallowed.
Also, I don't think the cache will avoid circular loop in case of circular references in the model itself once it's resolved.

Reply:
Yea, it's disallowed because if done in a certain way, you get a loop. The cache works as a mechanism to avoid chasing the tail if the reference is already resolved.

Reply:
I'm not sure I follow. I guess the goal of our resolver is to allow us to map this OAS to an API Gateway API, not necessarily properly resolve the schema? We don't need a valid OAS schema after resolving? Technically, I think it's still valid, it follows the retrieval URI spec: https://json-schema.org/understanding-json-schema/structuring.html#retrieval-uri

If AWS maps the Model this way in their `requestModels` and `requestResponses`, we can take this shortcut to allow us to map it in an easier way to what we need to create the API?
Sorry, I might be missing something here, genuine question.

Reply:
> AWS does not resolve the Model at all as shown by the snapshot
>      "house": {
>          "$ref": "<model-base-url>/restapis/<rest-id:1>/models/House"
>       }

Having a reference or the inline method is valid. So it doesn't matter what AWS does, in fact, what does it mean? Did you export  and that's how it came out?

Reply:
No, I'm talking to how it maps a schema in `components/schema` to a Model, and how it manages those references inside the Model, see https://stackoverflow.com/questions/37823112/how-do-i-reference-one-model-from-another-model-using-aws-api-gateway

Reply:
After thinking about this, we should try to import an API with no Model defined in `definitions/schema`, but instead directly set the schema. I wonder what comes out of this, because AWS would lack the schema name. I think all of this discussion can actually be solved by this test?

Reply:
To quickly come back on this, if you don't specify a $ref in `responses/<statusCode>/schema` field, AWS will automatically create the Model for you. 
Basically done the following (simplified):
```json
{
  "paths": "/pet",
  "post": {
    "responses": {
      "200": {
        "description": "Successful operation",
        "schema": {
          "type": "object",
          "properties": {
            "pet": {
              "$ref": "#/definitions/Pet"
            },
            "message": {
              "type": "string"
            }
          }
        },
        "headers": {
          "Access-Control-Allow-Origin": {
            "type": "string",
            "description": "URI that may access the resource"
          }
        }
      }
    }
  }
}
```

And AWS created the follow Model (normally json encoded Schema field): 
```json
{
  "name": "MODEL03987f",
  "contentType": "application/json",
  "schema": {
    "type": "object",
    "properties": {
      "pet": {
        "$ref": "https://apigateway.amazonaws.com/restapis/<api-id>/models/Pet"
      },
      "message": {
        "type": "string"
      }
    }
  }
}
```

Seems like we're still missing that feature, I'll note it and add it later on!